x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/ClientTransformIndexer.java

```diff
@@ -57,21 +64,24 @@
 
 class ClientTransformIndexer extends TransformIndexer {
 
+    private static final TimeValue PIT_KEEP_ALIVE = TimeValue.timeValueSeconds(30);
     private static final Logger logger = LogManager.getLogger(ClientTransformIndexer.class);
 
     private final Client client;
     private final AtomicBoolean oldStatsCleanedUp = new AtomicBoolean(false);
 
     private final AtomicReference<SeqNoPrimaryTermAndIndex> seqNoPrimaryTermAndIndex;
+    private PointInTimeBuilder pit;
```

Comment:
does `pit` and `disablePit` need to be `volatile`? They are accessed from separate threads in different execution paths.

Reply:
pit is always accessed by the indexer thread, even the `onStop` call originates from the indexer, not from the `_stop` transport if that's what you mean